### PR TITLE
Add the possibility to exclude users or groups from node sources

### PR DESCRIPTION
**Pull request by mbenguig**

### CODE REVIEW

File: common/common-client/src/main/java/org/ow2/proactive/permissions/PrincipalPermission.java [hunk 1]
```diff
@@ -151,11 +211,41 @@ public void add(Permission permission) {
...
-        for (Permission p : permissions) {
-            if (p.implies(permission)) {
-                return true;
+
+        if (!(permission instanceof PrincipalPermission)) {
+            return false;
+        }
+
+        PrincipalPermission permissionToRespect = (PrincipalPermission) permission;
+        if (permissionToRespect.principals.isEmpty())
```
Comment (Mykhailenko) [inline common/common-client/src/main/java/org/ow2/proactive/permissions/PrincipalPermission.java]: could you please put `{`

File: common/common-client/src/main/java/org/ow2/proactive/permissions/PrincipalPermission.java [hunk 2]
```diff
@@ -151,11 +211,41 @@ public void add(Permission permission) {
...
+
+        int nbPrincipals = permissionToRespect.principals.size();
+        long nbExcludedPrincipals = permissionToRespect.principals.stream()
+                                                                  .filter(c -> c instanceof NotIdentityPrincipal)
+                                                                  .count();
+
+        if (nbExcludedPrincipals == 0) {
+            // Give access if INCLUSION permissions to respect include/implie one of the permission requiring access
+            // ex: "Bob" or "GroupBob" get access if "Bob" or "GroupBob" is part of permissions to respect
+            for (Permission p : permissions) {
```
Comment (Mykhailenko) [inline common/common-client/src/main/java/org/ow2/proactive/permissions/PrincipalPermission.java]: you can use `Stream::anyMatch` method

File: common/common-client/src/main/java/org/ow2/proactive/permissions/PrincipalPermission.java (hunk 1 above)
Comment (mbenguig) [reply common/common-client/src/main/java/org/ow2/proactive/permissions/PrincipalPermission.java]: sure

File: common/common-client/src/main/java/org/ow2/proactive/permissions/PrincipalPermission.java (hunk 2 above)
Comment (mbenguig) [reply common/common-client/src/main/java/org/ow2/proactive/permissions/PrincipalPermission.java]: sure

**Pull request merged**